drape_frontend/stylist.cpp

```diff
@@ -318,20 +318,16 @@ void CaptionDescription::ProcessMainTextType(drule::text_type_t const & mainText
   if (m_houseNumber.empty())
     return;
 
-  if (mainTextType == drule::text_type_housenumber)
+  if (mainTextType == drule::text_type_housenumber || mainTextType == drule::text_type_name &&
+      (m_mainText.empty() || m_houseNumber.find(m_mainText) != std::string::npos))
   {
-    /// @todo this code path is never used, probably need to have e.g. "text: housenumber" in styles for it.
     m_mainText.swap(m_houseNumber);
-    m_houseNumber.clear();
```

Comment:
If we don't care what will be left in m_houseNumber, better to write here and below:
```
m_mainText = std::move(m_houseNumber);
```